codec-http2/src/main/java/io/netty/handler/codec/http2/UniformStreamByteDistributor.java

```diff
@@ -139,7 +148,11 @@ void updateStreamableBytes(int newStreamableBytes, boolean hasFrame) {
                 streamableBytes = newStreamableBytes;
                 totalStreamableBytes += delta;
             }
-            if (hasFrame) {
+            // We should queue this state if there is a frame. We don't want to queue this frame if the window
+            // size is <= 0 and we are writing this state. The rational being we already gave this state the chance to
+            // write, and if there were empty frames the expectation is they would have been sent. At this point there
+            // must be a call to updateStreamableBytes for this state to be able to write again.
+            if (hasFrame && (!writting || windowSize > 0)) {
```

Comment:
`pendingBytes == 0` should be accounted for by `hasFrame`. `windowSize == 0` is permitted if we are not writing. The rational being we allow it to be added to the queue to write 1 time, but if it is 0 during/after we write then we don't add it again (until updateStreamableBytes is called again).

Am I missing something?